app/cdap/components/GridTable/index.tsx

```diff
@@ -57,6 +61,22 @@ export default function GridTable() {
     },
   ]);
 
+  const { dataprep } = DataPrepStore.getState();
+  const [isFirstWrangle, setIsFirstWrangle] = useState(false);
+  const [connectorType, setConnectorType] = useState<string | null>(null);
```

Comment:
this file needs to useDispatch - there are way too many useStates here

Reply:
sorry i meant useReducer

Reply:
Fixed